bin/propolis-server/src/lib/vm/state_driver.rs

```diff
@@ -145,46 +145,53 @@ where
                 // the migration task so that reset doesn't overwrite the
                 // state written by migration.
                 self.reset_vcpus();
-                *next_lifecycle = Some(
-                    match self.migrate_as_target(
-                        migration_id,
-                        task,
-                        start_tx,
-                        command_rx,
-                    ) {
-                        Ok(_) => {
-                            LifecycleStage::NotStarted(StartupStage::Migrated)
-                        }
-                        Err(_) => LifecycleStage::NotStarted(
+                match self.migrate_as_target(
+                    migration_id,
+                    task,
+                    start_tx,
+                    command_rx,
+                ) {
+                    Ok(_) => {
+                        // Publish the "migrated" state before signaling that
+                        // migration is finished. Without this, there's a race
```

Comment:
I'm having a bit of trouble parsing this sentence. It might help to spell out more concretely what "signaling that migration is finished" means here.